local/src/main/java/com/kdjj/local/FileSaveHelper.kt

```diff
@@ -32,14 +34,39 @@ class FileSaveHelper @Inject constructor(
         byteArray: ByteArray,
         fileName: String
     ): Result<String> = withContext(Dispatchers.IO) {
-        val filePath = "$fileDir/$fileName.jpeg"
         var fos: FileOutputStream? = null
         try {
+            val filePath = "$fileDir/${fileName}.png"
             fos = FileOutputStream(filePath)
-            fos.write(byteArray)
+            val bitmap = convertByteArrayToBitmap(byteArray)
+            bitmap.compress(Bitmap.CompressFormat.PNG, 85, fos)
             Result.success(filePath)
         } catch (e: Exception) {
+            e.printStackTrace()
             Result.failure(e)
         }
     }
+
+    private fun convertByteArrayToBitmap(byteArray: ByteArray): Bitmap{
+        val options = BitmapFactory.Options()
+        options.inSampleSize = 4
+        val bitmap = BitmapFactory.decodeByteArray(byteArray, 0, byteArray.size)
+
+        val width = 300
+        val height = 300
+
+        var bmpWidth = bitmap.width.toFloat()
+        var bmpHeight = bitmap.height.toFloat()
+
+        if (bmpWidth > width) {
+            val scale = width / (bmpWidth / 100)
+            bmpWidth *= (scale / 100)
+            bmpHeight *= (scale / 100)
+        } else if(bmpHeight > height){
+            val scale = width / (bmpHeight / 100)
+            bmpWidth *= (scale / 100)
+            bmpHeight *= (scale / 100)
+        }
```

Comment:
```suggestion
        val standard = 300

        val width = bitmap.width.toFloat()
        val height = bitmap.height.toFloat()

        val (scaledWidth, scaledHeight) = if (oldWidth > standard || oldHeight > standard) {
            if (oldWitdh > oldHeight) {
                standard to (standard * height / width)
            } else {
                (standard * width / height) to standard
            }
        } else {
            width to height
        }
```

이렇게 바꾸면 조금 더 가독성이 좋을 것 같아요!